src/nvim/eval.h

```diff
@@ -3,6 +3,47 @@
 
 #include "nvim/profile.h"
 
+// All user-defined functions are found in this hashtable.
+hashtab_T func_hashtab;
+
+// Structure to hold info for a user function.
+typedef struct ufunc ufunc_T;
+
+struct ufunc {
+  int          uf_varargs;      //< variable nr of arguments
+  int          uf_flags;
+  int          uf_calls;        //< nr of active calls
+  garray_T     uf_args;         //< arguments
+  garray_T     uf_lines;        //< function lines
+  int          uf_profiling;    //< TRUE when func is being profiled
+  // Profiling the function as a whole.
+  int          uf_tm_count;     //< nr of calls
+  proftime_T   uf_tm_total;     //< time spent in function + children
+  proftime_T   uf_tm_self;      //< time spent in function itself
+  proftime_T   uf_tm_children;  //< time spent in children this call
+  // Profiling the function per line.
+  int         *uf_tml_count;    //< nr of times line was executed
+  proftime_T  *uf_tml_total;    //< time spent in a line + children
+  proftime_T  *uf_tml_self;     //< time spent in a line itself
+  proftime_T   uf_tml_start;    //< start time for current line
+  proftime_T   uf_tml_children; //< time spent in children for this line
+  proftime_T   uf_tml_wait;     //< start wait time for current line
+  int          uf_tml_idx;      //< index of line being timed; -1 if none
+  int          uf_tml_execed;   //< line being timed was executed
+  scid_T       uf_script_ID;    //< ID of script where function was defined,
+                                //  used for s: variables
+  int          uf_refcount;     //< for numbered function: reference count
+  char_u       uf_name[1];      //< name of function (actually longer); can
+                                //  start with <SNR>123_ (<SNR> is K_SPECIAL
+                                //  KS_EXTRA KE_SNR)
+};
+
+// From user function to hashitem and back.
+ufunc_T dumuf;
```

Comment:
Could be `static` (because only used to determine a distance in `ufunc_T`) or `extern` (needs an extra definition).